packages/inventory/src/containers/InventoryAddEdit/index.tsx

```diff
@@ -0,0 +1,274 @@
+import { fetchSettings } from './utils';
+import { Setting } from '../../components/InventoryItemForm';
+import { sendErrorNotification } from '@opensrp/notifications';
+import { getFetchOptions, HTTPError } from '@opensrp/server-service';
+import { RouteComponentProps } from 'react-router';
+import {
+  fetchLocationUnits,
+  getLocationUnitById,
+  LocationUnit,
+  locationUnitsReducerName,
+  locationUnitsReducer,
+} from '@opensrp/location-management';
+import {
+  INVENTORY_DONORS,
+  INVENTORY_SERVICE_POINT_PROFILE_VIEW,
+  INVENTORY_UNICEF_SECTIONS,
+  OPENSRP_ENDPOINT_GET_INVENTORIES,
+  OPENSRP_ENDPOINT_LOCATION,
+  OPENSRP_PRODUCT_CATALOGUE,
+  ROUTE_PARAM_INVENTORY_ID,
+  ROUTE_PARAM_SERVICE_POINT_ID,
+} from '../../constants';
+import {
+  fetchInventories,
+  getInventoriesByIdsFactory,
+  Inventory,
+  inventoryReducer,
+  inventoryReducerName,
+} from '../../ducks/inventory';
+import { InventoryItemForm, defaultInitialValues } from '../../components/InventoryItemForm';
+import { ProductCatalogue } from '@opensrp/product-catalogue';
+
+/** register reducers */
+reducerRegistry.register(locationUnitsReducerName, locationUnitsReducer);
+reducerRegistry.register(inventoryReducerName, inventoryReducer);
+
+/** Route params */
+interface RouteParams {
+  [ROUTE_PARAM_SERVICE_POINT_ID]: string;
+  [ROUTE_PARAM_INVENTORY_ID]?: string;
+}
+
+/** interface component props */
+export interface InventoryAddEditProps extends RouteComponentProps<RouteParams> {
+  openSRPBaseURL: string; // OpenSRP API base URL
+  servicePointProfileURL: string; // route to service point profile to redirect to after successful
+  // submission or clicking on cancel button to abort and exit
+  fetchLocationUnitsCreator: typeof fetchLocationUnits; // redux action to fetch location units
+  fetchInventoriesCreator: typeof fetchInventories; // redux action to fetch inventories
+  servicePoint: LocationUnit | null; // service point to add/edit inventory item
+  inventory?: Inventory | null; // inventory item to edit
+  customFetchOptions?: typeof getFetchOptions; // custom OpenSRPService class options
+}
+
+/** default component props */
+export const defaultInventoryAddEditProps = {
+  openSRPBaseURL: '',
+  servicePointProfileURL: INVENTORY_SERVICE_POINT_PROFILE_VIEW,
+  fetchLocationUnitsCreator: fetchLocationUnits,
+  fetchInventoriesCreator: fetchInventories,
+  servicePoint: null,
+};
+
+const InventoryAddEdit: React.FC<InventoryAddEditProps> = (props: InventoryAddEditProps) => {
+  const {
+    openSRPBaseURL,
+    servicePointProfileURL,
+    fetchLocationUnitsCreator,
+    fetchInventoriesCreator,
+    servicePoint,
+    inventory,
+    match,
+    customFetchOptions,
+  } = props;
+  const [UNICEFSections, setUNICEFSections] = React.useState<Setting[]>([]);
+  const [donors, setDonors] = React.useState<Setting[]>([]);
+  const [products, setProducts] = React.useState<ProductCatalogue[]>([]);
```

Comment:
should we be using the store here?

Reply:
@p-netm I decide to use local storage to address the stale data issue